docs/learn/provider_specific_features.md

```diff
@@ -47,11 +47,52 @@ if tool := response.tool:
 # > The Name of the Wind by Patrick Rothfuss
 ```
 
+Under the hood, Mirascope generates a JSON Schema for the `FormatBook` tool based on its attributes and the `OpenAIToolConfig`. This schema is then used by OpenAI's API to ensure the model's output strictly adheres to the defined structure.
+
+For comparison, here's how you might implement this using the official OpenAI SDK:
+
+```python
+from openai import OpenAI
+import json
+
+client = OpenAI()
+
+def format_book(title: str, author: str) -> str:
+    return f"{title} by {author}"
+
+function_json = {
```

Comment:
this example is not correct (primarily "functions" are deprecated). here are their docs on structured outputs: https://platform.openai.com/docs/guides/structured-outputs

note that the endpoint they show is beta, so you can still use the official endpoint manually. i think we should show using the official endpoint first (i.e. using `json_schema` with `strict=True`) and then showing their beta endpoint that enables using Pydantic more directly